File: outputs/2025-07-30_16-45-42-655837/test3.py
# Unit test result: ❌ FAILED (see log for details)
from enum import Enum, auto

class test3:

    class Priority(Enum):
        BAIXA = auto()
        MEDIA = auto()
        ALTA = auto()

    class Task:
        descricao: str
        prioridade: 'test3.Priority'

        def __init__(self, descricao: str, prioridade: 'test3.Priority'):
            self.descricao = descricao
            self.prioridade = prioridade

        def __str__(self) -> str:
            return f'[{self.prioridade.name}] {self.descricao}'

    @staticmethod
    def main(args: list[str]):
        tarefas: list['test3.Task'] = []
        continuar: bool = True
        while continuar:
            print('\n1. Adicionar tarefa')
            print('2. Listar tarefas')
            print('3. Sair')
            choice_str = input('Escolha uma opção: ')
            try:
                opcao = int(choice_str)
            except ValueError:
                print('Opção inválida. Por favor, insira um número.')
                continue
            match opcao:
                case 1:
                    desc = input('Descrição da tarefa: ')
                    prio_input = input('Prioridade (BAIXA, MEDIA, ALTA): ').upper()
                    prio: 'test3.Priority'
                    try:
                        prio = test3.Priority[prio_input]
                    except KeyError:
                        print('Prioridade inválida. Tarefa ignorada.')
                        continue
                    tarefas.append(test3.Task(desc, prio))
                    print('Tarefa adicionada.')
                case 2:
                    if not tarefas:
                        print('Nenhuma tarefa.')
                    else:
                        print('\nTarefas:')
                        for t in tarefas:
                            print(f'- {t}')
                case 3:
                    continuar = False
                    print('A sair...')
                case _:
                    print('Opção inválida.')
if __name__ == '__main__':
    test3.main([])

File: uploads/2025-07-30_16-45-42-655837/test3.java
package test_files.test3;
import java.util.ArrayList;
import java.util.Scanner;

public class test3 {

    enum Priority {
        BAIXA, MEDIA, ALTA
    }

    static class Task {
        String descricao;
        Priority prioridade;

        Task(String descricao, Priority prioridade) {
            this.descricao = descricao;
            this.prioridade = prioridade;
        }

        public String toString() {
            return "[" + prioridade + "] " + descricao;
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        ArrayList<Task> tarefas = new ArrayList<>();
        boolean continuar = true;

        while (continuar) {
            System.out.println("\n1. Adicionar tarefa");
            System.out.println("2. Listar tarefas");
            System.out.println("3. Sair");
            System.out.print("Escolha uma opção: ");
            int opcao = scanner.nextInt();
            scanner.nextLine(); // consumir newline

            switch (opcao) {
                case 1:
                    System.out.print("Descrição da tarefa: ");
                    String desc = scanner.nextLine();
                    System.out.print("Prioridade (BAIXA, MEDIA, ALTA): ");
                    String prioInput = scanner.nextLine().toUpperCase();
                    Priority prio;
                    try {
                        prio = Priority.valueOf(prioInput);
                    } catch (IllegalArgumentException e) {
                        System.out.println("Prioridade inválida. Tarefa ignorada.");
                        break;
                    }
                    tarefas.add(new Task(desc, prio));
                    System.out.println("Tarefa adicionada.");
                    break;
                case 2:
                    if (tarefas.isEmpty()) {
                        System.out.println("Nenhuma tarefa.");
                    } else {
                        System.out.println("\nTarefas:");
                        for (Task t : tarefas) {
                            System.out.println("- " + t);
                        }
                    }
                    break;
                case 3:
                    continuar = false;
                    System.out.println("A sair...");
                    break;
                default:
                    System.out.println("Opção inválida.");
            }
        }

        scanner.close();
    }
}
